R/hydromad.stats.R

```diff
@@ -441,6 +454,13 @@ buildCachedObjectiveFun <-
     "U1" = function(Q, X, ..., U) cor(head(Q - X, -1), tail(U, -1), use = "complete"),
     "r.sq.vartd" = function(Q, X, ..., event) {
       nseVarTd(Q, X, event, ...)
+    },
+    "KGE" = function(Q, X, ...) {
+      1 - sqrt(
+        (cor(X, Q, use = "complete") - 1)^2 +
+          (mean(X, na.rm = TRUE) / mean(Q, na.rm = TRUE) - 1)^2 +
+          (sd(X, na.rm = TRUE) / sd(Q, na.rm = TRUE) - 1)^2
+      )
```

Comment:
Just had a thought - the mean here will likely be computed on a different set of observations because Q will have NAs and X won't.
It might be better to use complete.cases first to identify the observations to omit for both Q and X? Could you have a look at rel.bias (I think) and see if that might be the right thing to do?

Reply:
Agreed. I've pushed a change that only uses complete cases.